### DISCUSSION TITLE
How can I version lock the installed just?

### DISCUSSION THREAD

**Discussion by ferologics** [casey/just#1548] [Q&A]
it'd be nice to have some kind of `.justversion` file in the root of the repo which determines what version the recipes expect, and if somebody has a wrong version it'll either auto-update or error out with `"pls update just to version {.justversion}"`

does just provide anything like this out of the box?

**Comment by runeimp** [comment]
That sounds cool and similar to many scripting language version management things. But generally speaking `just` will complain fairly loudly if you try to use a feature it doesn't recognize. The user upgrades and gets the latest version and you're typically good to go. Though I really do like the `.just-version` type file concept.

**Comment by airtonix** [comment]
just use [asdf](https://asdf-vm.com/) and [asdf-just](https://github.com/ggilmore/asdf-just)

this shouldn't be the responsibility of just.
